src/Features/Core/Portable/Completion/Providers/TypeImportCompletionItem.cs

```diff
@@ -0,0 +1,113 @@
+                 inlineDescription: containingNamespace,
+                 useEditorCompletionItemCache: true);
+        }
+
+        public static string GetContainingNamespace(CompletionItem item)
+        {
+            return item.Properties.TryGetValue(ContainingNamespaceName, out var containingNamespace)
+                ? containingNamespace
+                : null;
+        }
+
+        public static async Task<CompletionDescription> GetCompletionDescriptionAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var metadataName = GetMetadataName(item);
+            if (!string.IsNullOrEmpty(metadataName))
+            {
+                var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+                var symbol = compilation.GetTypeByMetadataName(metadataName);
+                if (symbol != null)
+                {
+                    var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+                    // We choose not to display the number of "type overloads" for simplicity. 
+                    // Otherwise, we need additonal logic to track internal and public visible
+                    // types separately, and cache both completion items.
+                    return await CommonCompletionUtilities.CreateDescriptionAsync(
+                        document.Project.Solution.Workspace,
+                        semanticModel,
+                        position: 0,
+                        symbol,
+                        overloadCount: 0,
+                        supportedPlatforms: null,
+                        cancellationToken).ConfigureAwait(false);
+                }
+            }
+
+            return null;
+        }
+
+        private static string GetInteger(int integer)
+        {
+            Debug.Assert(integer > 0);
+            return (integer <= s_IntegerOneToNine.Length)
+                ? s_IntegerOneToNine[integer - 1]
+                : integer.ToString(CultureInfo.InvariantCulture);
+        }
+
+        private static string GetAritySuffix(int arity)
+        {
+            Debug.Assert(arity > 0);
+            return (arity <= s_aritySuffixesOneToNine.Length)
+                ? s_aritySuffixesOneToNine[arity - 1]
+                : string.Concat(GenericTypeNameManglingString, arity.ToString(CultureInfo.InvariantCulture));
+        }
+
+        private static string GetFullyQualifiedName(string namespaceName, string typeName)
+            => namespaceName.Length == 0 ? typeName : namespaceName + "." + typeName;
+
+        private static string ComposeAritySuffixedMetadataName(string name, int arity)
+            => arity == 0 ? name : name + GetAritySuffix(arity);
+
+        private static string GetMetadataName(CompletionItem item)
+        {
+            if (item.Properties.TryGetValue(ContainingNamespaceName, out var containingNamespace))
```

Comment:
this seems more generally useful function. there are many times I guess people want to save symbol as string and get it back from GetTypeByMetadataName

but also, have you tried SymbolKey rather than you building metadata name yourself? is SymbolKey too heavy for your case? since description is lazy and created only when explicitly asked, SymbolKey might work just fine for you I think though.

Reply:
OK, I will look into that

Reply:
note: SymbolKey would be more heavyweight here.  I would def just stick with metadataname here.